tls/s2n_prf.c

```diff
@@ -157,6 +157,10 @@ static int s2n_sslv3_prf(struct s2n_connection *conn, struct s2n_blob *secret, s
 
         struct s2n_hash_state *md5 = workspace;
         POSIX_GUARD(s2n_hash_reset(md5));
+        /* FIPS specifically allows MD5 for the legacy PRF */
+        if (s2n_is_in_fips_mode() && conn->actual_protocol_version < S2N_TLS12) {
```

Comment:
Do you need the lines on 126 if you have these new lines? I just realized you're only starting to use md5 on line 164, maybe it's not necessary for line 126.

Reply:
Yes, thank you for pointing it out!